LatinumNetwork/mosaic
# [DAS-32] Add Mosaic to the Dashboard

## Purpose/Description:
Install mosaic into our dashboard and make sure things work correctly.
- Added the ability to export MUI components & MUI utilities from Mosaic. However, if we make a Button component, it will override MUI’s Button component.
- Adjusted peer dependencies to fix errors related to missing dependencies.
- Adjusted typography to due missing the component prop.
- Adjusted package.json, peerDependencies are controlled by users of mosaic, and devDependencies are needed to run the library locally with no issues. I had to add missing libs.
- the component Button needs to have access to the ref for it to work correctly, adjusted accordingly.

## Jira Link:

[DAS-32](https://collagegroup.atlassian.net/browse/DAS-32)

## Test Plan:

-   **Test Scenario 1:** Check if Mosaic packages correctly

    -   **Steps:**
        1. Run `npm install` in the app directory
        2. Run `npm run rollup` and verify there are no errors when packaging the application
    -   **Expected Result:** Rollup should package with no errors. Should look like this. 
    
![image](https://github.com/LatinumNetwork/mosaic/assets/1088165/9842e6f2-8fc4-4166-a8e2-15a48fbd7968)




[DAS-32]: https://collagegroup.atlassian.net/browse/DAS-32?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
